tracer/src/Datadog.Trace/Debugger/Instrumentation/Registry/EverGrowingRegistry.cs

```diff
@@ -0,0 +1,108 @@
+// <copyright file="EverGrowingRegistry.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Datadog.Trace.Debugger.Helpers;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger.Instrumentation.Registry
+{
+    /// <summary>
+    /// Holds a registry of indexed items that grow sparsely.
+    /// It utilizes a generic type <typeparamref name="TPayload"/> to define the type of items stored in the registry.
+    /// </summary>
+    internal abstract class EverGrowingRegistry<TPayload>
```

Comment:
Does the name `Registry` really make sense here? This is ultimately a collection data-type, that we are using a registry of sorts I suppose. Maybe `EverGrowingCollection`?

Reply:
Done in f2e39f19694d2e9a44c777ae15e0fa38a3d73526.